x/batch/msg_test.go

```diff
@@ -1,42 +1,53 @@
 package batch_test
 
 import (
-	"errors"
 	"testing"
 
 	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/weavetest/assert"
 	"github.com/iov-one/weave/x/batch"
-	. "github.com/smartystreets/goconvey/convey"
-	"github.com/stretchr/testify/mock"
 )
 
-func TestMsg(t *testing.T) {
-	Convey("Test Validate", t, func() {
-		msg := &mockMsg{}
-		Convey("Test happy flow", func() {
-			msg.On("MsgList").Return(make([]weave.Msg, 10), nil)
-			So(batch.Validate(msg), ShouldBeNil)
-			msg.AssertExpectations(t)
-		})
+func TestMsgValidate(t *testing.T) {
+	specs := map[string]struct {
+		msg batch.Msg
+		err *errors.Error
+	}{
+		"Happy flow": {
+			msg: &mockMsg{list: make([]weave.Msg, 10)},
```

Comment:
Shouldn't validation fail if the input is a list of `nil` messages?

Reply:
Nope, because the messages are actually coming from check/deliver mocks. This is by design.